mlrun/features.py

```diff
@@ -108,6 +108,14 @@ def __init__(
         self.aggregate = aggregate
         self.origin = None  # used to link the feature to the feature set origin (inside vector.status)
         self._validator = validator
+        # bypass DB bug - self.value_type have to be set when saving fs
+        if not self.value_type:
+            if isinstance(self._validator, MinMaxValidator):
+                self.value_type = "int"
+            elif isinstance(self._validator, MinMaxLenValidator):
+                self.value_type = "list"
+            else:  # including regex validator
+                self.value_type = "str"
```

Comment:
I like the big head in getting the type, but I'm not sure about it as it seems you may bypass a bug that is worth to correct from the root. @assaf758  take a look at the comment here.

Reply:
Yes, i agree with @guy1992l  that it's worth further investigation.
Are you sure it's a DB bug? ie is value_type really optional? If it's not optional I would simply raise an exception if it's not specified.

Reply:
I revert it and will open new bug on @assaf758 team.